src/HashingLogic.ts

```diff
@@ -104,7 +104,8 @@ export const verifyMerkleProof = (
   root: Buffer
 ): boolean => {
   // Should not succeed with all empty arguments
-  if (!proof.length || !targetNode || !root) {
+  // Proof can be empty if single leaf tree
+  if (!targetNode.toString() || !root.toString()) {
```

Comment:
This line seems odd... Why are you calling `toString` on a buffer object and using a `boolean` logic check on that string output?

Reply:
Checking if it’s empty. When I passed in a buffer derived from an empty string and checked if it was empty it didn’t work using if !targetNode. There’s probably a buffer function I could use instead.

Reply:
“Didn’t work” meaning the empty string buffer returned as not null

Reply:
I see, so does it return `''`? Would it be more correct / clearer if we checked for that instead?

Reply:
Another thing to consider is keeping the null checks in addition to the empty buffer check. This will handle js clients better.